src/modules/headset/oculus_mobile.c

```diff
@@ -456,6 +466,17 @@ void bridgeLovrInit(BridgeLovrInitData *initData) {
   LOG("\n BRIDGE INIT COMPLETE\n");
 }
 
+#define growAndCopyArray(OUT, IN, DATA, TYPE) \
+  { \
+      size_t arraySize = IN->members * sizeof(TYPE); \
+      if (IN->members > OUT.members) { \
+        free(OUT.DATA); \
+        OUT.DATA = malloc(arraySize); \
+      } \
+      OUT.members = IN->members; \
+      memcpy(OUT.DATA, IN->DATA, arraySize); \
+  }
+
```

Comment:
This is ugly and I hate it. I'm not sure I have a better way of doing this in C. The problem is the bridge has to convey 3 lists, one a list of strings (which never changes for a particular frame), one a list of poses (which changes every frame), one a list of confidence values (which changes every frame). I'm not sure what the best way to shuttle that data around in C is so I wound up creating 3 "list" structs, another thing I don't like but couldn't think of an obvious better option. Maybe I should be using arr_t here, I don't know. 

The "bridge" mechanism is definitely turning out to be a bigger inconvenience than I thought when I originally came up with it.

Reply:
Could it be a fixed-size array?  The number of bones is unlikely to change very often (I haven't seen a hand tracking SDK with more than 32, so far), and maybe the tradeoff of "may need to increase size limit later" is worth it compared to "need to deal with dynamic sizes/allocations".

Reply:
That would simplify things, but the problem with a fixed-size array is I think it would mean (because of the union) that we'd be paying to copy space for the fixed-size array even when using controllers (the typical case).